lua/plenary/path.lua

```diff
@@ -314,6 +314,47 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+function Path:rename(opts)
+  opts = opts or {}
+  if not opts.new_name or opts.new_name == "" then
+    print("Please provide the new name!")
+    return false
+  end
+
+  -- handles `.`, `..`, `./`, and `../`
```

Comment:
If you want to handle stuff like `../` you can do `vim.loop.fs_realpath` but the new_file doesn't exist so you might have to pop the last bit.
Because it might be useful to do `name.lua -> ../name.lua` or is moved not handled in rename. I haven't check uv docs

Reply:
I've added the ability to move the file to the parent directory by prefixing the filename with `../`. I didn't test it for Windows though.

Reply:
Thanks i need to think about some of and maybe play around with it for a minute or so :rofl: Doing this this weekend. Promise :)